src/services/wallet-service/wallet-service.ts

```diff
@@ -76,7 +84,7 @@ export class WalletService implements IWalletService {
 
     getCurrencyValue(currency: string) {
         return new Promise((resolve, reject) => {
```

Comment:
Instead of creating a new promise you can also use the `toPromise` function which turns a Observable into a Promise